test/libsolidity/ASTJSONTest.cpp

```diff
@@ -131,93 +131,106 @@ TestCase::TestResult ASTJSONTest::run(ostream& _stream, string const& _linePrefi
 	}
 	c.setSources(sources);
 	c.setEVMVersion(solidity::test::CommonOptions::get().evmVersion());
-	if (c.parse())
-		c.analyze();
-	else
+
+	if (!c.parse())
 	{
 		SourceReferenceFormatterHuman formatter(_stream, _formatted, false);
 		for (auto const& error: c.errors())
 			formatter.printErrorInformation(*error);
 		return TestResult::FatalError;
 	}
 
+	c.analyze();
+
+	bool resultsMatch = runTest(
+		m_expectation,
+		m_result,
+		sourceIndices,
+		c,
+		false,
+		"",
+		_stream,
+		_linePrefix,
+		_formatted
+	);
+
+	resultsMatch = runTest(
+		m_expectationLegacy,
+		m_resultLegacy,
+		sourceIndices,
+		c,
+		true,
+		"legacy",
+		_stream,
+		_linePrefix,
+		_formatted
+	) && resultsMatch;
+
+	return resultsMatch ? TestResult::Success : TestResult::Failure;
+}
+
+bool ASTJSONTest::runTest(
+	string& _expectation,
+	string& _result,
```

Comment:
Do we need these two parameters? I see that you pass private members. You can distinguish which set of parameters to use by checking `_legacy` bool param.

Reply:
Sure, this might work at this point, but it reduces the modularity a lot, if you have a look at the PR that builds on that you see what I mean: 

https://github.com/ethereum/solidity/pull/9364/commits/13638f46a26f17df6ffa564d2c154d3fca11769a#diff-4296f70f815b6207bf7ae36fa1ef6a2bqfqf0-